ai2thor/controller.py

```diff
@@ -519,7 +521,24 @@ def key_sort_func(scene_name):
 
         self.server.send(dict(action='Reset', sceneName=scene, sequenceId=0))
         self.last_event = self.server.receive()
-        
+
+        # update the initialization parameters
+        init_params = init_params.copy()
+
+        # width and height are updates in 'ChangeResolution', not 'Initialize'
+        if ('width' in init_params and  init_params['width'] != self.width) or (
+            'height' in init_params and init_params['height'] != self.height
+        ):
+            if 'width' in init_params:
+                self.width = init_params['width']
+                del init_params['width']
```

Comment:
Is the reason for deleting the param here to avoid forwarding them to Unity?

Reply:
Yeah. `width` and `height` aren't parameters within `controller.step('Initialize', ...)`.

Also, a copy of `init_params` is made, so deleting something should not be noticeable by the user.